internal/resourcestore/resourcestore.go

```diff
@@ -34,6 +37,7 @@ type Resource struct {
 	watchers []chan struct{}
 	stale    bool
 	name     string
+	stage    string
```

Comment:
Can we make this a first class type or enum?

Reply:
I had named strings a bit ago, but I think this is clearer and makes the source of the name closer to where it's relevant